### PR TITLE
gateway: Enhance the help output for gateway

### LINKED ISSUE #4196: azure-gateway: help backend needs to be expanded
help BACKEND needs to be expanded like USAGE and FLAG section
in BACKEND we need to explain corresponding endpoints.

### PR DESCRIPTION
**Pull request by krishnasrinivas**
## Description

Enhance the help output for gateway.

## Motivation and Context
Fixes #4196 

## How Has This Been Tested?
Manually.

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My code follows the code style of this project.
- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

File: cmd/gateway-main.go [hunk 1]
```diff
@@ -36,36 +36,30 @@ USAGE:
 FLAGS:
   {{range .VisibleFlags}}{{.}}
   {{end}}{{end}}
+BACKEND:
+  azure: Microsoft Azure Blob Storage. Default ENDPOINT is https://core.windows.net
```
Comment (nitisht) [inline cmd/gateway-main.go:40]: The URL **https://core.windows.net/** is not accessible on browser. Is that expected?

**Review by harshavardhana — APPROVED** [review]

File: cmd/gateway-main.go [hunk 2]
```diff
@@ -36,36 +36,30 @@ USAGE:
...
 ENVIRONMENT VARIABLES:
   ACCESS:
      MINIO_ACCESS_KEY: Username or access key of your storage backend.
      MINIO_SECRET_KEY: Password or secret key of your storage backend.
 
 EXAMPLES:
   1. Start minio gateway server for Azure Blob Storage backend.
+      $ export MINIO_ACCESS_KEY=myaccountname
+      $ export MINIO_SECRET_KEY=myaccountkey
       $ {{.HelpName}} azure
```
Comment (nitisht) [inline cmd/gateway-main.go]: Will it be better to use `azure_account_name` and `azure_account_key`?

**Comment by harshavardhana** [comment]
ping @krishnasrinivas @NitishT has some comments..

File: cmd/gateway-main.go (hunk 2 above)
Comment (krishnasrinivas) [reply cmd/gateway-main.go]: changed it to azureaccountname and azureaccountkey

File: cmd/gateway-main.go (hunk 1 above)
Comment (krishnasrinivas) [reply cmd/gateway-main.go:40]: Since the final url looks like azureaccountname.blob.core.windows.net which is resolvable it is fine

**Review by nitisht — APPROVED** [review]

**Pull request merged**